app/utility/base_knowledge_svc.py

```diff
@@ -108,7 +110,12 @@ async def _update_relationship(self, criteria, updates):
         for match in matches:
             for k, v in updates.items():
                 if getattr(match, k, False):
```

Comment:
hmm.. would it be good to also change the line above to compare against `None` here just in case we add additional attributes to the object that could evaluate to `false` in truth value testing?

Also, this logic appears that it will only apply updates if an object already has the attribute but any new attributes will not be added from updates if they don't already exist on the object.

Reply:
I can get behind swapping out `False`, but I don't want to check against `None` because that's the returned value if an attribute isn't initialized (but exists). I'm going to put in a wildcard that should account for this - we don't want people to put new attributes into the object that don't exist, just update existing ones that may not be initialized.

Reply:
ok, and that sounds good, I was not too sure if we wanted new attribute creation or not.

And as for the conditional as long as the value matches the specific case that it does not exist and not evaluate from a stored value that should cover that issue.

Reply:
and I see the update and that looks good 👍